diff --git a/node_modules/web3-providers-ws/lib/helpers.js b/node_modules/web3-providers-ws/lib/helpers.js
index 4a5e5b7..96a87a8 100644
--- a/node_modules/web3-providers-ws/lib/helpers.js
+++ b/node_modules/web3-providers-ws/lib/helpers.js
@@ -20,10 +20,14 @@ if (isNode || isRN) {
     }
 }
 else {
-    _btoa = btoa.bind(window);
-    helpers = function (url) {
-        return new URL(url);
-    };
+    if (typeof(window) !== 'undefined') {
+        _btoa = btoa.bind(window);
+
+        helpers = function (url) {
+           return new URL(url);
+       };
+    }
+    
 }
 module.exports = {
     parseURL: helpers,
